### PR TITLE
Make CI 3x faster

### PR DESCRIPTION
**Pull request by vicsn**
## Motivation

Already approved and tested here: https://github.com/ProvableHQ/snarkVM/pull/8

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -743,6 +743,36 @@ jobs:
           workspace_member: synthesizer/program
           cache_key: snarkvm-synthesizer-program-cache
 
+  synthesizer-program-integration-instruction-is:
+    docker:
+      - image: cimg/rust:1.76.0 # Attention - Change the MSRV in Cargo.toml and rust-toolchain as well
+    resource_class: << pipeline.parameters.2xlarge >>
```
Comment (zosorock) [inline .circleci/config.yml]: we need to change these from `2xlarge` to `twoxlarge` - here and in other locations (lines 759 and 769)

File: .circleci/config.yml [hunk 2]
```diff
@@ -743,6 +743,36 @@ jobs:
...
+    steps:
+      - run_serial:
+          flags: instruction::is --test '*'
+          workspace_member: synthesizer/program
+          cache_key: snarkvm-synthesizer-program-cache
+
+  synthesizer-program-integration-instruction-equal:
+    docker:
+      - image: cimg/rust:1.76.0 # Attention - Change the MSRV in Cargo.toml and rust-toolchain as well
+    resource_class: << pipeline.parameters.2xlarge >>
```
Comment (zosorock) [inline .circleci/config.yml]: we need to change these from `2xlarge` to `twoxlarge`

File: .circleci/config.yml [hunk 3]
```diff
@@ -743,6 +743,36 @@ jobs:
...
+    steps:
+      - run_serial:
+          flags: instruction::equal --test '*'
+          workspace_member: synthesizer/program
+          cache_key: snarkvm-synthesizer-program-cache
+
+  synthesizer-program-integration-instruction-commit:
+    docker:
+      - image: cimg/rust:1.76.0 # Attention - Change the MSRV in Cargo.toml and rust-toolchain as well
+    resource_class: << pipeline.parameters.2xlarge >>
```
Comment (zosorock) [inline .circleci/config.yml]: we need to change these from `2xlarge` to `twoxlarge`

**Review by zosorock — CHANGES_REQUESTED** [review]

File: synthesizer/program/tests/instruction/assert.rs [hunk 4]
```diff
@@ -28,7 +28,7 @@ use synthesizer_process::{Process, Stack};
 type CurrentNetwork = MainnetV0;
 type CurrentAleo = AleoV0;
 
-const ITERATIONS: usize = 100;
+const ITERATIONS: usize = 25;
```
Comment (gluax) [inline synthesizer/program/tests/instruction/assert.rs:31]: These types of tests should be avoided in CI, where we iterate and do random values. These randoms breaks unit and integration test philosophy, and also cause the CI time to increase. Unit tests should try to cover all edge cases, while random/fuzz tests are best done on a dedicated machine off CI. If a random/fuzz test does find something you would then convert it to a unit test.

**Review by gluax — APPROVED** [review]
Still approving for urgency, but how testing is being done should be reconsidered.

**Review by evanmarshall — COMMENTED** [review]
It's not clear to me why any of the values initially existed and why this new value is lower but still sufficiently acceptable. 

Any reason we shouldn't just set them to `1` or `5` or `10`?

**Review by Meshiest — APPROVED** [review]

**Review by voximity — APPROVED** [review]

**Review by zosorock — APPROVED** [review]

**Comment by vicsn** [comment]
> Any reason we shouldn't just set them to 1 or 5 or 10?

No valid reason, agree with @gluax ' [comment](https://github.com/AleoNet/snarkVM/pull/2484#discussion_r1633980016) and its worth investigating a refactor in the future. Somewhat related: there are some heavy `#[ignore]`'d tests which could potentially also fit in a separate fuzzing / CI / integration testing flow.
